app/services/persister_task_service.rb

```diff
@@ -0,0 +1,54 @@
+class PersisterTaskService
+  def initialize(options)
+    @options = JSON.parse(options).deep_symbolize_keys
+    @upload_task = Task.find(@options[:category])
+  end
+
+  def process
+    @task = FullRefreshPersisterTask.create!(opts)
+    create_kafka_event
+
+    self
+  end
+
+  def create_kafka_event
```

Comment:
@hsong-rh Should this be normalized into a separate module so that it can be called from multiple places. by passing in 

1. service
2. event
3. payload

Maybe something like this
https://github.com/RedHatInsights/sources-api/blob/6cb38a375101a5feaaac40d07453b048323960d9/lib/sources/api/events.rb#L4

But add the service also in the argument list as topic_name since it is a Kafka topic.